src/Components/Directory.Build.targets

```diff
@@ -30,4 +41,56 @@
                               PackagePath="buildTransitive\$(TargetFramework)\$(PackageId).targets" />
     </ItemGroup>
   </Target>
+
+  <PropertyGroup>
+    <ConfigurationSchemaGeneratorPath>$(ArtifactsBinDir)ConfigurationSchemaGenerator\$(Configuration)\$(NetCurrent)\ConfigurationSchemaGenerator.dll</ConfigurationSchemaGeneratorPath>
+
+    <GeneratedConfigurationSchemaOutputPath>$(IntermediateOutputPath)ConfigurationSchema.json</GeneratedConfigurationSchemaOutputPath>
+  </PropertyGroup>
+  
+  <Target Name="GenerateConfigurationSchema"
+          DependsOnTargets="CoreGenerateConfigurationSchema;
+                            CopyNewConfigurationSchema;
+                            CompareConfigurationSchema" />
+  
+  <Target Name="CoreGenerateConfigurationSchema"
+          Inputs="$(ConfigurationSchemaGeneratorPath);
+                  @(IntermediateAssembly);
+                  @(ReferencePathWithRefAssemblies)"
+          Outputs="$(GeneratedConfigurationSchemaOutputPath)">
+    <PropertyGroup>
+
+      <GeneratorCommandLine>"$(DotNetTool)" exec $(ConfigurationSchemaGeneratorPath)</GeneratorCommandLine>
+      <GeneratorCommandLine>$(GeneratorCommandLine) --input "@(IntermediateAssembly)"</GeneratorCommandLine>
+      <GeneratorCommandLine>$(GeneratorCommandLine) --reference "@(ReferencePathWithRefAssemblies, '" "')"</GeneratorCommandLine>
+      <GeneratorCommandLine>$(GeneratorCommandLine) --output "$(GeneratedConfigurationSchemaOutputPath)"</GeneratorCommandLine>
+    </PropertyGroup>
+    
+    <Exec Command="$(GeneratorCommandLine)" />
+
+    <ItemGroup>
+      <FileWrites Include="$(GeneratedConfigurationSchemaOutputPath)" Condition="Exists('$(GeneratedConfigurationSchemaOutputPath)')"/>
+    </ItemGroup>
+  </Target>
+
+  <Target Name="CopyNewConfigurationSchema"
+          Condition="'$(ConfigurationSchemaExists)' != 'true' OR '$(UpdateConfigurationSchema)' == 'true'">
+
+    <Copy SourceFiles="$(GeneratedConfigurationSchemaOutputPath)"
+          DestinationFiles="$(ConfigurationSchemaPath)" />
+
+  </Target>
+
+  <Target Name="CompareConfigurationSchema"
+          Condition="Exists('$(ConfigurationSchemaPath)') AND Exists('GeneratedConfigurationSchemaOutputPath')">
+
+    <PropertyGroup>
+      <CurrentConfigurationSchemaFileContent>$([System.IO.File]::ReadAllText('$(ConfigurationSchemaPath)'))</CurrentConfigurationSchemaFileContent>
+      <GeneratedConfigurationSchemaFileContent>$([System.IO.File]::ReadAllText('$(GeneratedConfigurationSchemaOutputPath)'))</GeneratedConfigurationSchemaFileContent>
```

Comment:
Probably a dumb question, but can encoding come into play here? I'm thinking the case where for some reason the checked in encoding happened to be on a different one than the generated one causing issues and errors for people. I suppose the answer is no, but asking just in case.

Reply:
This reads the files into a .NET `string` and compares them, it doesn't compare byte-by-byte of the files. So unless the strings are different, this won't fail.